### PR TITLE
docs - use resource groups to limit greenplum_fdw concurrency

### PR DESCRIPTION
**Pull request by lisakowen**
in this PR:  add a topic named "About Using Resource Groups to Limit Concurrency" to the greenplum_fdw module docs.  also changed some master refs -> coordinator.

questions:  are the CREATE RESOURCE GROUP statements correct and reasonable for this scenario?  is there any other info that would be helpful to the user?

doc review site link (behind vpn):  https://docs-staging.vmware.com/en/draft/VMware-Greenplum/gp2gp-resgroups/greenplum-database/GUID-ref_guide-modules-greenplum_fdw.html#about-using-resource-groups-to-limit-concurrency

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md [hunk 1]
```diff
@@ -184,6 +184,63 @@ These parameters are likely to be less problematic than `search_path`, but can b
...
+1. Create a dedicated resource group to manage resources for these users:
+
+    ```
+    CREATE RESOURCE GROUP rg_gpcluster2_users with (concurrency=2, cpu_max_percent=20);
+    ALTER ROLE gpcluster2_users RESOURCE GROUP rg_gpcluster2_users;
+    ```
+
+    When cluster 2 is configured as described below, this resource group manages the local resources on cluster 1 used by all queries that are initiated by cluster 2 `greenplum_fdw` users.
+
+(Initiating) cluster 2 configuration:
```
Comment (zxuejing) [inline gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md:208]: Hi @lisakowen 
The original task is how to control the remote concurrency which initiate by local cluster using greenplum_fdw.
So I think we can only introduce the configuration of remote cluster.
And local cluster user should map to remote cluster which in dedicated resouce group.

File: gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md [hunk 2]
```diff
@@ -184,6 +184,63 @@ These parameters are likely to be less problematic than `search_path`, but can b
 
 Do not override this behavior by changing the session-level settings of these parameters; that is likely to cause `greenplum_fdw` to malfunction.
 
+## <a id="resgroups"></a>About Using Resource Groups to Limit Concurrency
+
+You can create dedicated users and resource groups to manage `greenplum_fdw` concurrency on both the local and remote Greenplum clusters. In the following example, cluster 2 (initiating) reads data from cluster 1 (remote).
+
+(Remote) cluster 1 configuration:
+
+1. Create a dedicated Greenplum Database user/role to represent the `greenplum_fdw` users on cluster 2 that initiate queries. For example, to create a role named `gpcluster2_users`:
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md:193]: i think having an identifier in addition to local and remote is required.  when we create the resource group on the remote cluster, saying "... to represent the greenplum_fdw users on the local cluster" does not make sense.  i updated the docs to use local/remote as the primary cluster reference and use 1 and 2 as the secondary reference.  let me know if you don't think this makes things clearer.

File: gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md [hunk 3]
```diff
@@ -184,6 +184,63 @@ These parameters are likely to be less problematic than `search_path`, but can b
...
+
+    ```
+    CREATE SERVER gpc1_testdb FOREIGN DATA WRAPPER greenplum_fdw
+        OPTIONS (host 'gpc1_coordinator', port '5432', dbname 'testdb', mpp_execute 'all segments', );
+    ```
+
+1. Map all local users of the `greenplum_fdw` foreign server to the remote role. For example, to map users of the `gpc1_testdb` server on cluster 1 to the `gpcluster2_users` role on cluster 2:
+
+    ```
+    CREATE USER MAPPING FOR CURRENT_USER SERVER gpc1_testdb
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md]: updated to include user mappings for two specific users.

File: gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md (hunk 1 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md]: i removed the resource group configuration on the local side and updated the descriptions.

File: gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md [hunk 4]
```diff
@@ -184,6 +184,55 @@ These parameters are likely to be less problematic than `search_path`, but can b
...
+    ```
+
+1. Create a dedicated resource group to manage resources for these users:
+
+    ```
+    CREATE RESOURCE GROUP rg_gpcluster2_users with (concurrency=2, cpu_max_percent=20);
+    ALTER ROLE gpcluster2_users RESOURCE GROUP rg_gpcluster2_users;
+    ```
+
+    When you configure the remote cluster as described below, the `rg_gpcluster2_users` resource group manages the resources used by all queries that are initiated by `gpcluster2_users`.
```
Comment (zxuejing) [inline gpdb-doc/markdown/ref_guide/modules/greenplum_fdw.html.md:206]: When you configure the remote cluster as described below
below-->above?

**Review by zxuejing — APPROVED** [review]

**Comment by lisakowen** [comment]
thank you for reviewing, @zxuejing .  is this information also applicable to greenplum_fdw for greenplum 6 (assuming that the CREATE RESOURCE GROUP command is adjusted for greenplum 6 syntax)?  if so, i will create a backport PR.

**Comment by zxuejing** [comment]
> thank you for reviewing, @zxuejing . is this information also applicable to greenplum_fdw for greenplum 6 (assuming that the CREATE RESOURCE GROUP command is adjusted for greenplum 6 syntax)? if so, i will create a backport PR.

Yes I think this is applicable to greenplum_fdw for  greenplum 6.

**Review by dyozie — APPROVED** [review]

**Pull request merged**